pkg/file/buffer_test.go

```diff
@@ -158,31 +150,24 @@ func TestCopyBuffer(t *testing.T) {
 			chunkPipe := file.NewChunkPipe()
 
 			// destination
-			sizeC := make(chan int)
-			dataC := make(chan []byte)
+			sizeC := make(chan int, 1)
+			dataC := make(chan []byte, 1)
 			go reader(t, readBufferSize, chunkPipe, sizeC, dataC)
 
 			// source
-			errC := make(chan error, 1)
+			errC := make(chan error, 16)
 			go func() {
 				src := bytes.NewReader(srcBytes)
 
 				buf := make([]byte, swarm.ChunkSize)
 				c, err := io.CopyBuffer(chunkPipe, src, buf)
-				if err != nil {
-					errC <- err
-				}
+				errC <- err
```

Comment:
It would be better to return here if the err != nil right? If not we will be sending a duplicate error on line 167. The Close can be done in a defer statement.

Reply:
`errC` will collect all errors, and fail test whenever error happens, thus there will not be second error because first error would end test. 

---

> The Close can be done in a defer statement.

I am not sure what Close?